tests/modules/cache-cpp/lib.cpp

```diff
@@ -7,27 +7,62 @@
 #include "bindings/wasi-cache.h"
 #include "bindings/test.h"
 
+// TODO: for some reason bindings are returning
+// -1 upon success instead of WASI_CACHE_ERROR_SUCCESS (0)
```

Comment:
Huh, this is weird.
The errors seem to be properly defined in the bindings:
```cpp
  typedef uint8_t wasi_cache_error_t;
  #define WASI_CACHE_ERROR_SUCCESS 0
  #define WASI_CACHE_ERROR_ERROR 1
```

Reply:
Right, I don’t know what’s happening here, it’s probably something in `wit-bindgen`… but every function ends with a line like: 

```c
return variant2.tag ? variant2.val.err : -1;
```

Reply:
Ok, I'll look into that at some point.
Thanks for the heads up!

Reply:
For some reason bindgen emits a hardcoded value here:
https://github.com/bytecodealliance/wit-bindgen/blob/e9c7c0a3405845cecd3fe06f3c20ab413302fc73/crates/gen-c/src/lib.rs#L1863

Reply:
@radu-matei figured out what is going on here, the nesting of errors was confusing me before. `WASI_CACHE_ERROR_SUCCESS` is only returned in the `Err(test::Error::Success)` case, it is explicitly _not_ the `Ok` case! On the other hand, the `Ok` case should probably have its own constant to compare to, which `wit-bindgen` does not currently create.